eventuate-adapter-vertx/src/main/scala/com/rbmhtechnology/eventuate/adapter/vertx/api/EventMetadata.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2015 - 2016 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate.adapter.vertx.api
+
+import java.util.{ Optional => JOption }
+
+import com.rbmhtechnology.eventuate.{ DurableEvent, EventsourcedActor, EventsourcedProcessor }
+import io.vertx.core.MultiMap
+import io.vertx.rxjava.core.{ MultiMap => RxMultiMap }
+
+import scala.compat.java8.OptionConverters._
+
+object EventMetadata {
+  import com.rbmhtechnology.eventuate.adapter.vertx.VertxExtensions._
+
+  private[vertx] val MessageProducer = "producer"
+  private[vertx] val VertxProducer = "vertx-producer"
+
+  object Headers_ {
+    val LocalSequenceNr = "localSequenceNr"
+    val LocalLogId = "localLogId"
+    val EmitterId = "emitterId"
+  }
+
+  // exposure through a value to make the headers accessible for the Java-API
+  val Headers = Headers_
+
+  /**
+   * Creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[Option]] containing the metadata if this message was sent by a Vert.x producer otherwise [[None]].
+   */
+  def apply(headers: MultiMap): Option[EventMetadata] =
+    headers.getAsOption(MessageProducer).filter(_ == VertxProducer)
+      .map { s =>
+        new EventMetadata(
+          headers.getOrElseThrow(Headers.LocalLogId),
+          headers.getOrElseThrow(Headers.LocalSequenceNr).toLong,
+          headers.getOrElseThrow(Headers.EmitterId))
+      }
+
+  /**
+   * Creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
+   * if this message originated from a Vert.x producer.
+   *
+   * @param headers Headers of the event bus message.
+   * @return An [[Option]] containing the metadata if this message was sent by a Vert.x producer otherwise [[None]].
+   */
+  def apply(headers: RxMultiMap): Option[EventMetadata] =
+    apply(headers.getDelegate.asInstanceOf[MultiMap])
+
+  /**
+   * Java-API that creates an instance of [[EventMetadata]] containing event metadata extracted from the headers of an event bus message
```

Comment:
It depends on whether we create a separate implemenation for the Java-API.
If we do not the factory method would be in the .japi-package whereas the returned instance of it would be in the .api-package which would be quite confusing.